lib/piggly/parser/grammar.tt

```diff
@@ -271,9 +271,21 @@ grammar Piggly
   end
 
   rule varDeclaration
+    varDeclarationMisc / varDeclarationCursor
+  end
+
+  rule varDeclarationMisc
     name:tIdentifier tSpace ( kwCONSTANT tSpace )?
-    type:tType ( tSpace  kwNOT tSpace kwNULL )?
-               ( tSpace? kwASSIGN tSpace? rval:expressionUntilSemiColon )?
+    type:tType ( tSpace  kwCOLLATE name:tIdentifier )?
+               ( tSpace  kwNOT tSpace kwNULL )?
+               ( tSpace? ( kwASSIGN / kwDEFAULT ) tSpace? rval:expressionUntilSemiColon )?
+    tSpace?
+    ';' tSpace?
+  end
+
+  rule varDeclarationCursor
+    name:tIdentifier tSpace ( kwNO tSpace kwSCROLL tSpace )? kwCURSOR tSpace
```

Comment:
According to [this document](https://www.postgresql.org/docs/current/plpgsql-cursors.html), `NO` is optional. So both `DECLARE foo SCROLL CURSOR ...` and `DECLARE foo NO SCROLL CURSOR ...` should be accepted. You should be able to change this to `( kwNO tSpace )?` I think.

Reply:
Sure.  Will do.
Also, I need to make at least one other slight tweak, so please hold off on merging this just yet.

Reply:
Ok, adjusted.  Please check out the current set of changes.